### PR TITLE
feat(tooltip): add support for background and font color overrides FE-3377

### PR DESCRIPTION
**Pull request by edleeks87**
### Proposed behaviour

Adds support to `Tooltip` for overriding background and font colour.
Adds support for leveraging the colour overrides in `Icon` and `Help`
Adds `word-break: break-word` to tooltip wrapper to prevent long content strings overflowing to fix `FE-2020`

### Current behaviour

No colour override support in `Tooltip`, `Icon` and `Help`
Long strings do not wrap in `Tooltip` wrapper

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### Additional context

image

image

image

Word wrap FE-2020 fix:
image

### Testing instructions

Stories will colour overrides have been added to `Tooltip`, `Icon` and `Help`

https://codesandbox.io/s/carbon-quickstart-forked-f8ist?file=/src/index.js

### CODE REVIEW

File: src/components/tooltip/tooltip.d.ts [hunk 1]
```diff
@@ -17,6 +17,8 @@ export interface TooltipProps {
     size?: 'medium' | 'large';
     isPartOfInput?: boolean;
     inputSize?: 'small' | 'medium' | 'large';
+    bgColor?: string;
+    fontColor: string;
```
Comment (DipperTheDan) [inline src/components/tooltip/tooltip.d.ts:21]: Need a `?` here as fontColor is optional.

**Review by DipperTheDan — APPROVED** [review]

**Review by mmaciorowski-sage — APPROVED** [review]

File: src/components/icon/icon.stories.mdx [hunk 2]
```diff
@@ -46,12 +46,14 @@ const MyComponent = () => (
 
 ### With tooltip
 The Icon supports rendering tooltips internally, just pass a string to `tooltipMessage` to enable this feature. Other props, 
-such as `tooltipPosition` and `tooltipVisible` are also surfaced.
+such as `tooltipPosition`, `tooltipVisible` , `tooltipBgColor` and `tooltipFontColor` are also surfaced.
 
 <Preview>
   <Story name="with tooltip">
```
Comment (sjohnson-sage) [inline src/components/icon/icon.stories.mdx:52]: We can disable this story for Chromatic

Comment (edleeks87) [reply src/components/icon/icon.stories.mdx:52]: I think they're disabled in the Meta of this aren't they?

File: src/components/tooltip/tooltip.stories.mdx [hunk 3]
```diff
@@ -206,6 +206,39 @@ or if no value is passed it will default to a black background.
   </Story>
 </Preview>
 
+### Color overrides
+The `Tooltip` background and font color can be overridden via the `bgColor` and `fontColor` props and accept any valid 
+css value. Any values passed to the `bgColor` prop will override the color applied by the `type`
+
+<Preview>
+  <Story name="color overrides">
```
Comment (sjohnson-sage) [inline src/components/tooltip/tooltip.stories.mdx:214]: We can disable this for Chromatic as well

File: src/components/icon/icon.stories.mdx (hunk 2 above)
Comment (sjohnson-sage) [reply src/components/icon/icon.stories.mdx:52]: So they are

**Review by sjohnson-sage — APPROVED** [review]

**Review by DipperTheDan — DISMISSED** [review]

File: src/components/help/help.stories.mdx [hunk 4]
```diff
@@ -51,6 +51,21 @@ Hover over the help component to see the different positions.
   </Story>
 </Preview>
 
+### Help with Tooltip color overrides
+The tooltip background and font colors can be overridden using the `tooltipBgColor` and `tooltipFontColor` props. These 
+props accept and valid css color value.
+
+<Preview>
+  <Story name="with Tooltip color overrides">
```
Comment (DlgSHi) [inline src/components/help/help.stories.mdx:59]: this story should be ignored in `chromatic` the same as `default` and `with icons`

**Review by DipperTheDan — APPROVED** [review]

**Review by sjohnson-sage — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**